zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -53,48 +81,58 @@ impl<F: Field> ExecutionGadget<F> for PushGadget<F> {
         //                           ▼                     ▼
         //   [byte31,     ...,     byte2,     byte1,     byte0]
         //
-        for idx in 0..32 {
+        // First is_pushed (resp. is_padding) will always be 1 (resp. 1)
+        let mut pu_prev = 1.expr();
+        let mut pa_prev = 1.expr();
+        for (idx, (pu, pa)) in is_pushed.iter().zip(is_padding.iter()).enumerate() {
             let byte = &value.limbs[idx];
             let index = cb.curr.state.program_counter.expr() + opcode.expr()
                 - (OpcodeId::PUSH1.as_u8() - 1 + idx as u8).expr();
-            if idx == 0 {
-                cb.opcode_lookup_at(index, byte.expr(), 0.expr())
-            } else {
-                cb.condition(selectors[idx - 1].expr(), |cb| {
-                    cb.opcode_lookup_at(index, byte.expr(), 0.expr())
-                });
-            }
-        }
 
-        for idx in 0..31 {
-            let selector_prev = if idx == 0 {
-                // First selector will always be 1
-                1.expr()
-            } else {
-                selectors[idx - 1].expr()
-            };
-            // selector can transit from 1 to 0 only once as [1, 1, 1, ...,
-            // 0, 0, 0]
+            // is_pushed can transit from 1 to 0 only once
+            // as 1 - [1, 1, 1, ..., 1, 0, 0, 0]
+            cb.require_boolean(
+                "Constrain is_pushed can only transit from 1 to 0",
+                pu_prev - pu.expr(),
+            );
+
+            // when is_pushed = 0, is_padding = 1
+            // otherwise, is_padding can transit from 1 to 0 only once
+            // as 1 - [1, 1, 0, ..., 0, 1, 1, 1] (out of bound)
+            // as 1 - [0, 0, 0, ..., 0, 1, 1, 1] (not out of bound)
             cb.require_boolean(
-                "Constrain selector can only transit from 1 to 0",
-                selector_prev - selectors[idx].expr(),
+                "Constrain is_padding can only transit from 1 to 0 when is_pushed = 1, else is 1",
```

Comment:
My impression from the specs was that both `is_padding` and `is_pushed` followed the pattern `[1 1 ... 0 0]`.  But here `is_padding` can be `[ 11 0 ... 0 1 1]`.  Could you comment on this difference?

Reply:
Sure! Does the explanation in the previous post satisfy you?

Reply:
I will chagne it back to 11000 as said previously